Re-edit of PGD - third set of edits (#5784)
EnterpriseDB/docs: product_docs/docs/pgd/5/durability/camo.mdx

For a CAMO partner to switch to ready, it needs to be connected, and the estimated catchup interval needs to drop below the `timeout` value of `TO ASYNC`. You can check the current readiness status of a CAMO partner with [`bdr.is_camo_partner_ready()`](/pgd/latest/reference/functions#bdris_camo_partner_ready), while [`bdr.node_replication_rates`](/pgd/latest/reference/catalogs-visible#bdrnode_replication_rates) provides the current estimate of the catchup time.

> The switch from CAMO-protected to asynchronous mode is only ever triggered by an actual CAMO transaction. This is true either because the commit exceeds the `timeout` value of `TO ASYNC` or, in case the CAMO partner is already known, disconnected at the time of commit. This switch is independent of the estimated catchup interval. If the CAMO pair is configured to require the current node to be the write lead of a group as configured through the `enable_proxy_routing` node group option. See [Commit scopes](commit-scopes) for syntax. This can prevent a split brain situation due to an isolated node from switching to asynchronous mode. If `enable_proxy_routing` isn't set for the CAMO group, the origin node switches to asynchronous mode immediately.

Author: This is a fragment: 
If the CAMO pair is configured to require the current node to be the write lead of a group as configured through the `enable_proxy_routing` node group option.
Can you figure out what was meant? 

Also, I was again not able to trace what "This" meant in the second-to-last sentence.

> <!-- Sentence fragment, this what? >

Author: ```suggestion
```

```
PGconn *conn = PQconnectdb(origin_dsn);
```

The process starts connecting to the origin node. Now enter the loop:

```
loop {
    PQexec(conn, "BEGIN");
```

Next, start the transaction and begin populating it with changes:

```
    PQexec(conn, "INSERT INTO ...");
    ...
```

Once you're done, you need to make a record of the local node id and the transaction id. Both are available as parameters.

```
    node_id = PQparameterStatus(conn, "bdr.local_node_id");
    xid = PQparameterStatus(conn, "transaction_id");
```

Now it's ready to try to commit.

```
    PQexec(conn, "COMMIT");
    if (PQresultStatus(res) == PGRES_COMMAND_OK)
        return SUCCESS;
```

If the result is `PGRES_COMMAND_OK`, that's good, and you can move on. But if it isn't, you need to use CAMO to track the transaction to completion. The first question to ask is, "Was the connection bad?"

```
    else if (PQstatus(res) == CONNECTION_BAD)
    {
```

If it was a bad connection, then you can check on the CAMO partner node to see if the transaction made it there.

```
        conn = PQconnectdb(partner_dsn);
        if (!connectionEstablished())
            panic();
```

> If you can't connect to the partner node, there's not a lot you can do. So we should panic, or take similar actions.

Author: I don't think "panic" is a technical term? I needed to change this to second person, but I wasn't sure if this was humor (which we shouldn't use) or what.

Reviewer: It is - panic means to dump core and exit with an error. for example in Go language it is a standard function to panic when an error condition can't be resolved.

Author: ```suggestion
If you can't connect to the partner node, there's not a lot you can do. In this case, panic, or take similar actions.
```

> <!-- surely not >

Author: ```suggestion
```

Take care to ensure that the application fetches the proper node id. When using session pooling, the client remains connected to the same node, so the node id remains constant for the lifetime of the client session. However, with finer-grained transaction pooling, the client needs to fetch the node id for every transaction, as in the example that follows.

A client that isn't directly connected to the PGD nodes might not even notice a failover or switchover. But it can always use the `bdr.local_node_id` parameter to determine the node it's currently connected to. In the crucial situation of a disconnect during COMMIT, the proxy must properly forward that disconnect as an error to the client applying the CAMO protocol.

For CAMO in `received` mode, a proxy that potentially switches between the CAMO pairs must use the `bdr.wait_for_camo_partner_queue` function to prevent stale reads.

## CAMO limitations

CAMO limitations are covered in [Durability limitations](limitations/#camo).

## Performance implications

> CAMO extends the Postgres replication protocol by adding a message round trip at commit. Applications have a higher

Reviewer: ```suggestion
CAMO extends the Postgres replication protocol by adding a message roundtrip at commit. Applications have a higher
```
